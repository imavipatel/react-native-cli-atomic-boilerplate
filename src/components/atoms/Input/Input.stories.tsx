import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';
import { Input } from './Input';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

export const actions = {
  onPinTask: action('onPinTask'),
  onArchiveTask: action('onArchiveTask'),
};
storiesOf('Input', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <Input title="Primary Input" />);
