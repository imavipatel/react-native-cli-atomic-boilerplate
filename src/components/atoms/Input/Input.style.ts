import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  inputContainer: {
    minWidth: 200,
    minHeight: 40,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: '#08949B',
    borderStyle: 'solid',
    borderRadius: 20,
    borderWidth: 2,
    padding: 12,
  },
});
