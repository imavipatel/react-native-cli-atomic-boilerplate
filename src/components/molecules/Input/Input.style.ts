import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  inputContainer: {
    display: 'flex',
    flexDirection: 'column',
  },
});
