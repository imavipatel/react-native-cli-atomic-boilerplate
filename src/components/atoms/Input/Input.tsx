import React from 'react';
import { KeyboardTypeOptions, TextInput } from 'react-native';
import { styles } from './Input.style';

interface InputProps {
  title: string;
  keyboardType?: KeyboardTypeOptions;
  secureTextEntry?: boolean;
}

export const Input: React.FC<InputProps> = props => {
  return (
    <TextInput
      style={styles.inputContainer}
      placeholder={props.title}
      keyboardType={props.keyboardType}
      secureTextEntry={props.secureTextEntry}
      testID="input"
    />
  );
};

Input.defaultProps = {
  title: 'Input Title',
};
