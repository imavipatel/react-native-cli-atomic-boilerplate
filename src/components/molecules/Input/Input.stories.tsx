import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Input } from './Input';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('Input', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <Input label="Primary Input" />);
