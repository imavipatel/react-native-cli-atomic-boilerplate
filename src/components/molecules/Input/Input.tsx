import React from 'react';
import { KeyboardTypeOptions, View } from 'react-native';

import { TextInput, Label } from '../../atoms';
import { styles } from './Input.style';

interface InputProps {
  label: string;
  keyboardType?: KeyboardTypeOptions;
  secureTextEntry?: boolean;
  testID: string;
}

export const Input: React.FC<InputProps> = props => {
  return (
    <View style={styles.inputContainer}>
      <Label>{props.label}</Label>
      <TextInput
        placeholder={props.label}
        keyboardType={props.keyboardType}
        secureTextEntry={props.secureTextEntry}
        testID={props.testID}
      />
    </View>
  );
};

Input.defaultProps = {
  label: 'Input Title',
};
