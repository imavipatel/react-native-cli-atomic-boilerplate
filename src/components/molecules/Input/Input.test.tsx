import React from 'react';
import { render } from '@testing-library/react-native';
import { Input } from './Input';

describe('TextInput', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(<Input testID="input" label="Primary Input" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render input', () => {
    const component = <Input testID="input" label="Primary Input" />;
    const { getByTestId } = render(component);
    const input = getByTestId('input');
    expect(input).toBeDefined();
  });

  it('should render correctly input', () => {
    const testID = 'input';
    const placeHolder = 'input placeholder';
    const keyboardType = 'email-address';

    const { getByTestId } = render(
      <Input testID="input" label={placeHolder} keyboardType={keyboardType} />,
    );

    const textInput = getByTestId(testID);
    expect(textInput.props.placeholder).toEqual(placeHolder);
    expect(textInput.props.keyboardType).toEqual(keyboardType);
  });

  it('should render input label', () => {
    const label = 'Primary Input';
    const component = <Input testID="input" label={label} />;
    const { getByText } = render(component);
    const input = getByText(label);
    expect(input).toBeDefined();
  });
});
