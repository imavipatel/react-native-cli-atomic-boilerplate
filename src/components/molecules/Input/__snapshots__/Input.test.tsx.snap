// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "height": 78,
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
    testID="label"
  >
    Primary Input
  </Text>
  <TextInput
    placeholder="Primary Input"
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#08949B",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 1,
        "fontSize": 16,
        "fontWeight": "600",
        "justifyContent": "center",
        "minHeight": 50,
        "minWidth": 200,
        "paddingLeft": 17,
        "paddingRight": 17,
      }
    }
    testID="input"
  />
</View>
`;
